src/main/java/com/harrison/reflections/domain/EmployeeRead.java

```diff
@@ -0,0 +1,94 @@
+package com.harrison.reflections.domain;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonIdentityReference;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+
+@Entity
+@Table(name = "employee")
+public class EmployeeRead implements Serializable {
+
+    private static final long serialVersionUID = -4172063317499569232L;
+
+    @Id
+    @Column(name="EMPLOYEE_ID")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long employeeId;
+    
+    @Column(name="FIRST_NAME")
+    private String firstName;
+    
+    @Column(name="LAST_NAME")
+    private String lastName;
+    
+    @ManyToOne(cascade={CascadeType.ALL})
+    @JoinColumn(name="manager_id")
+    @JsonIdentityInfo(
+            generator = ObjectIdGenerators.PropertyGenerator.class, 
+            property = "employeeId"
+    )
+    @JsonIdentityReference(alwaysAsId = true)
+    private EmployeeRead manager;
+
+    @OneToMany(cascade = CascadeType.ALL)
+    @JoinColumn(name = "manager_id")
+    private Set<EmployeeRead> subordinates = new HashSet<EmployeeRead>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.38.14.86:9000/sonar/coding_rules#rule_key=squid%3AS2293)